### ISSUE TITLE
Loading recursion depth

### ISSUE THREAD

**Issue by Kamik423** [mitsuhiko/pluginbase#16]
Would it be possible to add a recursion depth for the search path?

```
my_program.py
plugins
    - plugin 1
        - foo.py
        - bar.py
    - plugin 2
        - foobar.py
```

Currently I am scanning plugins with `os.listdir`, and searchpathing each. Having a recursion depth would be great.

**Comment by zeroSteiner** [comment]
So with `plugins/` as the source for you're plugins you'd like to be able to load `foo` as a plugin found in `plugins/plugin 1`? You can't really do this right now. You could specify a plugin source (see [step 2](http://pluginbase.pocoo.org/)) with both `plugins/plugin 1` or `plugins/plugin 2` as directories.

Additionally if your plugins are just large in general and you'd like to keep them broken out into different files you can actually make the plugin itself a package with a structure like:
```
my_program.py
plugins
    - plugin_1
        - __init__.py
        - foo.py
        - bar.py
    - plugin_2
        - __init__.py
        - foobar.py
```

And the `__init__.py` files can import from the modules it contains with a relative import or IIRC an absolute one.

**Comment by Kamik423** [comment]
this is supposed to be suggestion (however I cannot set flags here), sorry for not making that clear.

I'd like there to be an option to do a deeper search when creating a plugin source. The way I want it is the way for example mods of many popular games work currently: You drop in a folder containing multiple scripts and other resources, and all of those get loaded.

I currently do this by:
```
self.plugin_source = self.plugin_base.make_plugin_source(
            searchpath=['plugins/{}'.format(plugin) for plugin in os.listdir('plugins')])
```

**Comment by zeroSteiner** [comment]
Ok, so in commit dcd30e63d52cfc4a6818bec676010aded5991cee I added a utility function called `get_searchpath`. It should meet your use case. You'd just need to update your example to:

```
self.plugin_source = self.plugin_base.make_plugin_source(
            searchpath=get_searchpath('plugins'), depth=3)
```
You can update the `depth` argument to whatever number you'd like. It defaults to `inf` and will recursively add all child directories it finds. There's also a `followlinks` argument you can set.

**Comment by zeroSteiner** [comment]
FYI, this is in the latest release available via PyPi now in `0.7`.
